code/modules/projectiles/projectile.dm

```diff
@@ -60,9 +62,15 @@
 	return ..()
 
 /obj/item/projectile/proc/Range()
-	range--
-	if(range <= 0 && loc)
-		on_range()
+    range--
+    if(damage > 0 && tile_dropoff)
+        damage = max(0, damage - tile_dropoff) // decrement projectile damage based on dropoff value for each tile it moves
+    if(stamina > 0 && tile_dropoff)

```

Comment:
this second check should be `if(stamina > 0 && tile_dropoff_s)` not `tile_dropoff`